radio/src/mixer.cpp

```diff
@@ -527,13 +527,19 @@ int getStickTrimValue(int stick, int stickValue)
     return 0;
 
   int trim = trims[stick];
-  if (stick == THR_STICK) {
-    if (g_model.thrTrim) {
-      int trimMin = g_model.extendedTrims ? 2*TRIM_EXTENDED_MIN : 2*TRIM_MIN;
-      trim = ((g_model.throttleReversed ? (trim+trimMin) : (trim-trimMin)) * (RESX-stickValue)) >> (RESX_SHIFT+1);
-    }
-    if (g_model.throttleReversed) {
+  uint8_t thrTrimSw;
+  if (g_model.thrTrimSw == 0) //this is needed for compatibility with existing models
+      thrTrimSw = MIXSRC_TrimThr - MIXSRC_FIRST_TRIM;
+    else if (g_model.thrTrimSw == MIXSRC_TrimThr - MIXSRC_FIRST_TRIM)
+      thrTrimSw = 0;
+    else
+      thrTrimSw = g_model.thrTrimSw;
```

Comment:
This is needed for compatibility with current models. May be removed with next eeprom update